src/bundles/curve/functions.ts

```diff
@@ -739,3 +793,49 @@ export function unit_line_at(t: number): Curve {
 export function arc(t: number): Point {
   return make_point(Math.sin(Math.PI * t), Math.cos(Math.PI * t));
 }
+
+/**
+ * Create a animation of curves using a curve generating function.
+ * @param duration The duration of the animation in seconds
+ * @param fps Framerate of the animation in frames per second
+ * @param drawer Draw function to the generated curves with
+ * @param func Curve generating function. Takes in a timestamp value and returns a curve
+ * @return Curve Animation
+ */
+export function animate_curve(
+  duration: number,
+  fps: number,
+  drawer: RenderFunction,
+  func: CurveAnimation
+): AnimatedCurve {
+  if ((drawer as any).is3D) {
```

Comment:
Most of the single line curly brace code is due to the stylistic choices enforced by prettier. For example, this line is broken into two by prettier. Since it's across multiple lines I use brackets